src/datamap.rs

```diff
@@ -8,103 +8,82 @@ pub struct ChunkDetails {
   pub hash: Vec<u8>,
   /// Pre encryption hash of chunk
   pub pre_hash: Vec<u8>,
-  /// size before encryption (compression alters this as well as any poss padding 
+  /// size before encryption (compression alters this as well as any poss padding
   /// depending on cipher used)
   pub source_size: u64
 }
 
 impl ChunkDetails {
   pub fn new() -> ChunkDetails {
-    ChunkDetails { chunk_num : 0, hash : Vec::new(), pre_hash : Vec::new(), source_size : 0 }
-  }
-}
-
-impl Clone for ChunkDetails {
-  fn clone(&self) -> ChunkDetails {
-    return ChunkDetails {
-      chunk_num: self.chunk_num,
-      hash: self.hash.to_vec(),
-      pre_hash: self.pre_hash.to_vec(),
-      source_size: self.source_size
-    };
+    ChunkDetails {
+      chunk_num: 0,
+      hash: vec![],
+      pre_hash: vec![],
+      source_size: 0
+    }
   }
 }
 
 /// Holds the infomation that required to recover the content of the encrypted file
 /// Depends on the file size, such info can be held as vector of ChunkDetail, or as raw data directly
-#[derive(PartialEq, Eq, PartialOrd, Ord)] 
+#[derive(PartialEq, Eq, PartialOrd, Ord)]
 pub enum DataMap {
-  /// if file was large enough (larger than 3072 bytes, 3 * MIN_CHUNK_SIZE) 
+  /// if file was large enough (larger than 3072 bytes, 3 * MIN_CHUNK_SIZE)
   /// this holds the list of chunks' info to decrypt it
   Chunks(Vec<ChunkDetails>),
-  /// very small files (less than 3072 bytes, 3 * MIN_CHUNK_SIZE) are put here in entirely 
+  /// very small files (less than 3072 bytes, 3 * MIN_CHUNK_SIZE) are put here in entirely
   Content(Vec<u8>),
   /// empty datamap
   None
-  }
+}
 
 impl DataMap {
   /// original size of file in datamap
-  pub fn len(&self)->u64 {
-    let size = 0u64;
+  pub fn len(&self) -> u64 {
     match *self {
-      DataMap::Chunks(ref chunks) => DataMap::chunks_size(chunks), 
-        DataMap::Content(ref content) => content.len() as u64, 
-        DataMap::None => 0u64
+      DataMap::Chunks(ref chunks) => DataMap::chunks_size(chunks),
+      DataMap::Content(ref content) => content.len() as u64,
+      DataMap::None => 0
```

Comment:
It's redundant as the type is inferred from both the other match arms and the return type.